docs/src/transforms.md

```diff
@@ -0,0 +1,246 @@
+# [Transforms](@id about-transforms)
+
+A `Transform` defines a transformation of data for feature engineering purposes.
+Some examples are scaling, periodic functions, linear combination, and one-hot encoding.
+
+```@meta
+DocTestSetup = quote
+    using DataFrames
+    using Dates
+    using FeatureTransforms
+end
+```
+
+## Defining a transform
+
+A `Transform` often has one or more parameters.
+For example, the following defines a squaring operation (i.e. raise to the power of 2):
+
+```julia-repl
+julia> p = Power(2);
+```
+
+## Methods to apply a transform
+
+Given some data `x`, there are three main methods to apply a transform.
+Firstly, it can be applied in a non-mutating fashion using `apply`:
+
+```jldoctest transforms
+julia> p = Power(2);
+
+julia> x = [1.0, 2.0, 3.0];
+
+julia> FeatureTransforms.apply(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+
+julia> x
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+
+Equivalently, the `Transform` object can be called directly on the data:
+
+```jldoctest transforms
+julia> p(x)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+Alternatively, the data can be mutated using the `apply!` method.
+
+!!! note
+
+    Not all `Transform` subtypes support mutation.
+
+```jldoctest transforms
+julia> FeatureTransforms.apply!(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+
+julia> x
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+A single `Transform` instance can be applied to different data types, with support for `AbstractArray`s and [`Table`s](https://github.com/JuliaData/Tables.jl).
+
+!!! note
+
+    Some `Transform` subtypes have restrictions on how they can be applied once constructed.
+    For instance, `MeanStdScaling` stores the mean and standard deviation of some data for specified dimensions or column names.
+    So `MeanStdScaling` should only be applied to the same data type and for the same dimensions or subset of column names specified in construction.
+
+## Applying to `AbstractArray`
+
+### Default
+
+Without specifying optional arguments, a `Transform` is applied to every element of an `AbstractArray` and in an element-wise fashion:
+
+```jldoctest transforms
+julia> M = [2 4; 1 5; 3 6];
```

Comment:
You might want to change this to a float array to avoid failing the 32 bit tests